File: scala/src/main/scala/tennis/TennisGame1.scala
package tennis


class TennisGame1 (val player1Name : String, val player2Name : String) extends TennisGame {
  var m_score1: Int = 0
  var m_score2: Int = 0

  def wonPoint(playerName : String) {
          if (playerName == "player1")
              m_score1 += 1
          else
              m_score2 += 1
  }

  /**
   *
   * @param score Int between 0 and 2
   * @return scoreConv
   * @author Robin CAVALIERI
   */
  def equalScore(score: Int) : String = {
    score match {
        case 0 =>  "Love-All"
        case 1 => "Fifteen-All"
        case 2 => "Thirty-All"
        case _ => "Deuce"
    }
  }

  /**
   *
   * @param minusResult
   * @return
   */
  def getMatchStatus(minusResult: Int) : String = {
    if (minusResult == 1) {
      "Advantage player1"
    }
    else if (minusResult == -1) {
      "Advantage player2"
    }
    else if (minusResult >= 2) {
      "Win for player1"
    }
    else {
      "Win for player2"
    }
  }

  /**
   *
   * @param scorePlayer1 Int between 0 and 2
   * @param scorePlayer2 Int between 0 and 2
   * @return score
   * @author Robin CAVALIERI
   */
  def winningScore(scorePlayer1: Int, scorePlayer2: Int) : String = {
    getMatchStatus(scorePlayer1-scorePlayer2)
  }

  /**
   *
   * @param score
   * @return
   */
  def convScore(score: Int): String = {
    score match {
      case 0 => "Love"
      case 1 => "Fifteen"
      case 2 => "Thirty"
      case 3 => "Forty"
    }
  }

  /**
   *
   * @param scorePlayer1 Int between 0 and 2
   * @param scorePlayer2 Int between 0 and 2
   * @return
   * @author Robin CAVALIERI
   */
  def diffScore(scorePlayer1: Int, scorePlayer2: Int): String = {
    convScore(scorePlayer1) + "-" + convScore(scorePlayer2)
  }

  /**
   *
   * @return
   */
  def calculateScore() : String = {

      if (m_score1 == m_score2)
      {
          equalScore(m_score1)
      }
      else if (m_score1>=4 || m_score2>=4)
      {
          winningScore(m_score1, m_score2)
      }
      else
      {
          diffScore(m_score1, m_score2)
      }
  }

}

File: python/src/TennisGame1.py
# -*- coding: utf-8 -*-

class TennisGame1:

    def __init__(self, player1Name, player2Name):
        self.player1Name = player1Name
        self.player2Name = player2Name
        self.p1points = 0
        self.p2points = 0

    def won_point(self, playerName):
        if playerName == self.player1Name:
            self.p1points += 1
        else:
            self.p2points += 1

    def equal_score(self):
        return {
            0: "Love-All",
            1: "Fifteen-All",
            2: "Thirty-All",
            }.get(self.p1points, "Deuce")

    def score_more_than_4(self):
        minusResult = self.p1points - self.p2points
        if minusResult == 1:
            result = "Advantage " + self.player1Name
        elif minusResult == -1:
            result = "Advantage " + self.player2Name
        elif minusResult >= 2:
            result = "Win for " + self.player1Name
        else:
            result = "Win for " + self.player2Name
        return result

    def diff_score(self):
        score_str_dico = {
                0: "Love",
                1: "Fifteen",
                2: "Thirty",
                3: "Forty",
            }
        return "-".join(map(score_str_dico.get, [self.p1points, self.p2points]))

    def score(self):
        if (self.p1points == self.p2points):
            return self.equal_score()
        elif (self.p1points >= 4 or self.p2points >= 4):
            return self.score_more_than_4()
        else:
            return self.diff_score()
